### PR TITLE
TMA-705 deprecated delete_projects and delete_extra param and added …

### PR DESCRIPTION
**Pull request by Hahihula**
…new delete_mode

### CODE REVIEW

File: lib/gooddata/lcm/actions/associate_clients.rb [hunk 1]
```diff
@@ -46,7 +49,21 @@ class AssociateClients < BaseAction
       ]
 
       class << self
+        DELETE_MODES = %w(
+          none
+          delete_projects
+          delete_extra
+        )
         def call(params)
+          if params.delete_projects && !params.delete_extra
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/associate_clients.rb]: Calls 'params.delete_extra' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/associate_clients.rb]: Calls 'params.delete_projects' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/associate_clients.rb]: Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/lcm/actions/associate_clients.rb [hunk 2]
```diff
@@ -46,7 +49,21 @@ class AssociateClients < BaseAction
...
+            GoodData.logger.warn("Parameter `delete_projects` is set to true and parameter `delete_extra` is set to false (which is default) this action will do nothing!")
+            return
+          end
+
+          unless DELETE_MODES.include?(params.delete_mode) || params.delete_mode.nil?
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/associate_clients.rb]: Calls 'params.delete_mode' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/associate_clients.rb]: Performs a nil-check (NilCheck)

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@ def check_params(specification, params)
             value = params.send(param_name)
             type = specification[param_name][:type]
             if value.nil? || (value.is_a?(String) && value.empty?)
-              if specification[param_name][:opts][:default]
+              if !specification[param_name][:opts][:default].nil?
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name]' 6 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name][:opts]' 5 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name][:opts][:default]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Refers to 'specification' more than self (maybe move it to another class?) (FeatureEnvy)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Performs a nil-check (NilCheck)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/170/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1006/ : SUCCESS in 4m 37s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1275/ : FAILURE in 4m 21s

**Pull request closed without merging**